cmd/tendermint/commands/lite.go

```diff
@@ -99,34 +104,91 @@ func runProxy(cmd *cobra.Command, args []string) error {
 
 	witnessesAddrs := strings.Split(witnessAddrsJoined, ",")
 
+	// if no witnesses were provided make sure we pass an empty string
+	if witnessAddrsJoined == "" {
+		witnessesAddrs = []string{}
+	}
+
 	db, err := dbm.NewGoLevelDB("light-client-db", home)
 	if err != nil {
 		return fmt.Errorf("can't create a db: %w", err)
 	}
 
+	if primaryAddr == "" {
+		var err error
+		primaryAddr, witnessesAddrs, err = checkForExistingProviders(db)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve primary or witness from db. Error: %w", err)
+		}
+		if primaryAddr == "" {
+			return errors.New("no primary address was provided nor found. Please provide a primary (using -p)." +
+				" Run the command: tendermint light --help for more information")
+		}
+	} else {
+		err := saveProviders(db, primaryAddr, witnessAddrsJoined)
+		if err != nil {
+			logger.Error("Unable to save primary and or witness addresses", "err", err)
+		}
+	}
+
+	trustLevel, err := tmmath.NewFraction(trustLevelStr)
```

Comment:
```suggestion
	trustLevel, err := tmmath.ParseFraction(trustLevelStr)
```

by analogy with `func ParseInt(s string, base int, bitSize int) (i int64, err error)`